rllib/examples/env/bandit_envs_recommender_system.py

```diff
@@ -1,166 +1,227 @@
+        if self.num_docs_in_db is not None:
+            if self.docs_db is None:
+                self.docs_db = [
+                    self._get_embedding() for _ in range(self.num_docs_in_db)
+                ]
+            self.currently_suggested_docs = [
+                self.docs_db[doc_idx].astype(np.float32)
+                for doc_idx in np.random.choice(
+                    self.num_docs_in_db,
+                    size=(self.num_docs_to_select_from,),
+                    replace=False,
+                )
+            ]
+        # Pick from an infinite pool of docs.
         else:
-            user_obs_space = spaces.Discrete(self.num_users)
-            return spaces.Dict(
-                {
-                    "user": user_obs_space,
-                    "item": item_obs_space,
-                    "item_id": item_ids_obs_space,
-                    "response": resp_space,
-                }
-            )
+            self.currently_suggested_docs = [
+                self._get_embedding() for _ in range(self.num_docs_to_select_from)
+            ]
 
-    def step(self, action):
-        # Action can be a single action or a slate depending on slate size
-        assert self.action_space.contains(
-            action
-        ), "Action cannot be recognized. Please check the type and bounds."
-
-        if self.slate_size == 1:
-            scores = self.item_pool.dot(self.user_embeddings[self.current_user_id])
-            reward = scores[action]
-            regret = np.max(scores) - reward
-            self.total_regret += regret
-
-            info = {"regret": regret}
-
-            self.current_user_id = np.random.randint(0, self.num_users)
-            self._gen_item_pool()
-
-            obs = {
-                "item": self.item_pool.astype(np.float32),
-                "item_id": self.item_pool_ids,
-                "response": [reward],
-            }
-            if self.num_users > 1:
-                obs["user"] = self.current_user_id
-            return obs, reward, True, info
-        else:
-            # TODO(saurabh3949):Handle slate recommendation using a click model
-            return None
+        doc = {str(i): d for i, d in enumerate(self.currently_suggested_docs)}
 
-    def reset(self):
-        self._sample_user()
-        self._gen_item_pool()
-        obs = {
-            "item": self.item_pool,
-            "item_id": self.item_pool_ids,
-            "response": [0] * self.slate_size,
+        if not response:
+            response = self.observation_space["response"].sample()
+
+        return {
+            "user": self.current_user.astype(np.float32),
+            "doc": doc,
+            "response": response,
         }
-        if self.num_users > 1:
-            obs["user"] = self.current_user_id
-        return obs
 
-    def render(self, mode="human"):
-        raise NotImplementedError
+
+# Because ParametricRecSys follows RecSim's API, we have to wrap it before
```

Comment:
Should we do this here?
Maybe just move this into the block below and register it under "ParametricRecSysEnvForBandits" or something like this.
Also, in the example script that actually uses this env, we can register it to show, how this is done.

Reply:
I really like the idea of registering the env in example script. good demo.
thanks.